docs/rst/tutorials/micro/custom_platforms/custom_platforms.rst

```diff
@@ -0,0 +1,341 @@
+              "cmake-args": [
+                  "-DENABLE_TESTING=OFF",
+                  "-DRCUTILS_NO_FILESYSTEM=ON",
+                  "-DRCUTILS_NO_THREAD_SUPPORT=ON",
+                  "-DRCUTILS_NO_64_ATOMIC=ON",
+                  "-DRCUTILS_AVOID_DYNAMIC_ALLOCATION=ON"
+              ]
+          },
+          "microxrcedds_client": {
+              "cmake-args": [
+                  "-DUCLIENT_PIC=OFF",
+                  "-DUCLIENT_PROFILE_UDP=OFF",
+                  "-DUCLIENT_PROFILE_TCP=OFF",
+                  "-DUCLIENT_PROFILE_DISCOVERY=OFF",
+                  "-DUCLIENT_PROFILE_SERIAL=OFF",
+                  "-UCLIENT_PROFILE_STREAM_FRAMING=ON",
+                  "-DUCLIENT_PROFILE_CUSTOM_TRANSPORT=ON"
+              ]
+          },
+          "rmw_microxrcedds": {
+              "cmake-args": [
+                  "-DRMW_UXRCE_MAX_NODES=1",
+                  "-DRMW_UXRCE_MAX_PUBLISHERS=5",
+                  "-DRMW_UXRCE_MAX_SUBSCRIPTIONS=5",
+                  "-DRMW_UXRCE_MAX_SERVICES=1",
+                  "-DRMW_UXRCE_MAX_CLIENTS=1",
+                  "-DRMW_UXRCE_MAX_HISTORY=4",
+                  "-DRMW_UXRCE_TRANSPORT=custom"
+              ]
+          }
+      }
+  }
+
+An example of a **CMake toolchain** for a Cortex-M3 platform can be:
+
+.. code-block::
+
+  set(CMAKE_SYSTEM_NAME Generic)
+  set(CMAKE_CROSSCOMPILING 1)
+  set(CMAKE_TRY_COMPILE_TARGET_TYPE STATIC_LIBRARY)
+
+  # SET HERE THE PATH TO YOUR C99 AND C++ COMPILERS
+  set(CMAKE_C_COMPILER arm-none-eabi-gcc)
+  set(CMAKE_CXX_COMPILER arm-none-eabi-g++)
+
+  set(CMAKE_C_COMPILER_WORKS 1 CACHE INTERNAL "")
+  set(CMAKE_CXX_COMPILER_WORKS 1 CACHE INTERNAL "")
+
+  # SET HERE YOUR BUILDING FLAGS
+  set(FLAGS "-O2 -ffunction-sections -fdata-sections -fno-exceptions -mcpu=cortex-m3 -nostdlib -mthumb --param max-inline-insns-single=500 -DF_CPU=84000000L" CACHE STRING "" FORCE)
+
+  set(CMAKE_C_FLAGS_INIT "-std=c11 ${FLAGS} -DCLOCK_MONOTONIC=0 -D'__attribute__(x)='" CACHE STRING "" FORCE)
+  set(CMAKE_CXX_FLAGS_INIT "-std=c++11 ${FLAGS} -fno-rtti -DCLOCK_MONOTONIC=0 -D'__attribute__(x)='" CACHE STRING "" FORCE)
+
+Once both files are ready, the micro-ROS library can be generated and built using the following command:
+
+.. code-block:: bash
+
+  ros2 run micro_ros_setup build_firmware.sh $(pwd)/my_custom_toolchain.cmake $(pwd)/my_custom_colcon.meta
+
+Creating a custom build script
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The procedure of creating a custom script that is able to generate a micro-ROS Client library highly depends on the approach taken.
+
+A basic understanding on how to proceed can be extracted analyzing the code of the `custom library generation script <https://github.com/micro-ROS/micro_ros_setup/blob/humble/config/generate_lib/generic/build.sh>`_ explained above.
+But in general the following points shall be taken into account:
+
+1. Create a micro-ROS development environment with the following packages in the correct branch:
```

Comment:
This is optional if we are in a sourced ROS2 environment right?

Reply:
A ROS 2 environment shall not be sourced in order to avoid generating Fast DDS type supports.

Note here: 43b29e82d8463c2b1a7c39d75ede487018b1cdee